### ISSUE TITLE
Bug on 3.0.0

### ISSUE THREAD

**Issue by arnauddorgans** [SDWebImage/SDWebImageSwiftUI#303]
Since 3.0.0 WebImage is not able to update its URL

on v2.2.7
<video src="https://github.com/SDWebImage/SDWebImageSwiftUI/assets/10015557/85b72f98-5ec6-4322-b5a2-79f31b90e04b">

on v3.0.0
<video src="https://github.com/SDWebImage/SDWebImageSwiftUI/assets/10015557/ea0aad65-9539-4b65-923a-1ae2e6f5503a">

Full code
```swift
struct ContentView: View {
    @State var isOn = false

    var url: URL? {
        if isOn {
            .init(string: "https://upload.wikimedia.org/wikipedia/commons/thumb/c/c1/Google_%22G%22_logo.svg/1024px-Google_%22G%22_logo.svg.png")
        } else {
            nil
        }
    }

    var body: some View {
        VStack {
            WebImage(url: url)
                .resizable()
                .scaledToFit()
                .frame(width: 50, height: 50)
            Button("Toggle") {
                isOn.toggle()
            }
        }
    }
}
```

I found a workaround by setting `.id(url)` but I don't think this is optimized

**Comment by arnauddorgans** [comment]
I have the same issue on `AnimatedImage` but only when switching between `nil` and non `nil` URL

**Comment by dreampiggy** [comment]
`nil` url only ? What's your SDWebImage version, please provide the full Pod version (upload your Podfile.lock)

A reproduce demo is better.

**Comment by dreampiggy** [comment]
> I found a workaround by setting .id(url) but I don't think this is optimized

@arnauddorgans  Actually, I compare the `2.x` and `master` branch and found this.

I guess this cause issue ? Why does we still need this identifier ? Is `nil` url special ?

<img width="1228" alt="截屏2024-03-18 16 30 23" src="https://github.com/SDWebImage/SDWebImageSwiftUI/assets/6919743/f6d02f3b-8354-4228-8300-73fe67c97d36">

**Comment by dreampiggy** [comment]
Fixed in [v3.0.1](https://github.com/SDWebImage/SDWebImageSwiftUI/releases/tag/3.0.1)

**Comment by arnauddorgans** [comment]
It works perfectly now, thanks! ❤️
